expression/column.go

```diff
@@ -191,6 +192,16 @@ func (col *CorrelatedColumn) resolveIndicesByVirtualExpr(_ *Schema) bool {
 	return true
 }
 
+// MemoryUsage return the memory usage of CorrelatedColumn
+func (col *CorrelatedColumn) MemoryUsage() (sum int64) {
+	if col == nil {
+		return
+	}
+
+	sum = col.Column.MemoryUsage() + int64(unsafe.Sizeof(col.Data))
```

Comment:
Better to implement `MemoryUsage` for `Datum` since it is commonly used in TiDB.